framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -119,41 +122,83 @@ std::shared_ptr<void> ServiceReferenceBasePrivate::GetService(BundlePrivate* bun
 
 InterfaceMapConstPtr ServiceReferenceBasePrivate::GetServiceInterfaceMap(BundlePrivate* bundle)
 {
+  /*
```

Comment:
Ah, I probably read or inferred that from the source code of a Java implementation. But I have thought about your use case as well and I do not see how we could differentiate such a case from a legitimate concurrent access which is not recursive. If you want to shoot yourself in the foot, you certainly can. Btw, the two Java implementations I looked at implemented this in different ways, none of them according to the spec as far as I can tell...